railties/lib/rails/application/configuration.rb

```diff
@@ -521,7 +521,8 @@ def secret_key_base
       end
 
       def secret_key_base=(new_secret_key_base)
-        if new_secret_key_base.is_a?(String) && new_secret_key_base.present?
+        if new_secret_key_base.nil? && generate_local_secret?
```

Comment:
Whenever I see an empty if branch like this, I assume it's a bug. Could just be me, but here's how I would write it to make it less confusing.

```suggestion
        if new_secret_key_base.nil? && generate_local_secret?
          @secret_key_base = nil # will be generated at runtime
```

Or even...

```suggestion
        if new_secret_key_base.nil? && generate_local_secret?
          # secret_key_base will be generated at runtime
```

Reply:
It also seems a bit strange that we check `generate_local_secret?` here when we already call before calling this method (line 515).

Reply:
Generating the local token when `new_secret_key_base` is nil can be pretty confusing as well.
Maybe we should output a warning message that the local token is used or generated?